crates/rover-client/src/blocking/client.rs

```diff
@@ -194,4 +232,75 @@ mod tests {
         let actual_error = handle_graphql_body_errors(errors).unwrap_err().to_string();
         assert_eq!(actual_error, expected_error);
     }
+
+    #[test]
+    fn test_successful_response() {
+        let server = MockServer::start();
+        let success_path = "/throw-me-a-frickin-bone-here";
+        let success_mock = server.mock(|when, then| {
+            when.method(POST).path(success_path);
+            then.status(200).body("I'm the boss. I need the info.");
+        });
+
+        let client = ReqwestClient::new();
+        let graphql_client = GraphQLClient::new(&server.url(success_path), client).unwrap();
+
+        let response = graphql_client.execute("{}", HeaderMap::new());
+
+        let mock_hits = success_mock.hits();
+
+        if mock_hits != 1 {
```

Comment:
Maybe assert rather than panic. It's easier to read.
Same goes for the other checks.